front/src/service/AuthenticationService.js

```diff
@@ -0,0 +1,51 @@
+import axios from 'axios';
+
+const API_URL = 'http://localhost:8080';
+export const USER_NAME_SESSION_ATTRIBUTE_NAME = 'authenticatedUser';
+
+class AuthenticationService {
+
+    executeAuthenticationService(username, password) {
+        return axios.get(`${API_URL}/auth`,
+            {headers: {authorization: this.createBasicAuthToken(username, password)}});
+    }
+
+    createBasicAuthToken(username, password) {
+        return 'Basic ' + window.btoa(username + ":" + password);
+    }
+
+    registerSuccessfulLogin(username, password) {
+        sessionStorage.setItem(USER_NAME_SESSION_ATTRIBUTE_NAME, username);
+        this.setupAxiosInterceptors(this.createBasicAuthToken(username, password));
+        document.getElementById("sign-in").hidden = true;
+        let loginLink = document.getElementById("login");
```

Comment:
why replace `login` with `logout`?
wouldn' it be better to hide/remove login component and create `logut`?

Reply:
maybe create third component "log-out" hodden be default and manage hiding those 3 components (login, logout, signin)